nuget/helpers/lib/NuGetUpdater/NuGetUpdater.Core/Updater/WebApplicationTargetsConditionPatcher.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System;
+using System.IO;
+using System.Linq;
+
+using Microsoft.Language.Xml;
+
+namespace NuGetUpdater.Core.Updater
+{
+    internal class WebApplicationTargetsConditionPatcher : IDisposable
+    {
+        private string? _capturedCondition;
+        private readonly XmlFilePreAndPostProcessor _processor;
+
+        public WebApplicationTargetsConditionPatcher(string projectFilePath)
+        {
+            _processor = new XmlFilePreAndPostProcessor(
+                getContent: () => File.ReadAllText(projectFilePath),
+                setContent: s => File.WriteAllText(projectFilePath, s),
+                nodeFinder: doc => doc.Descendants()
```

Comment:
Should this return multiple nodes? Sometimes people have multiple imports of the same file (accidentally or otherwise).

Reply:
To fix the currently open issue I'd prefer to leave it as-is, but regarding your below comment, if/when this needs to support multiples, we could update the post-processor to also accept an index/instance to update.  We can't keep the actual object, or count on file location, though, because between the `PreProcess` and `PostProcess` methods we invoke NuGet which can do just about anything to the structure of the `.csproj`.